### ISSUE TITLE
libhwmon depends on libdl (glibc) on Linux

### ISSUE THREAD

**Issue by reagentoo** [ethereum-mining/ethminer#625]
```
[100%] Linking CXX executable ethminer
/usr/lib/gcc/x86_64-pc-linux-gnu/7.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../libhwmon/libhwmon.a(wrapnvml.cpp.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
/lib64/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

**Comment by waltercool** [comment]
I can confirm this, adding dl to target_link_libraries fix the problem for me.

**Comment by smurfy** [comment]
@waltercool can you please provide me with the info what you exactly changed to fix the problem?
I can incorporate the change in my hwmon improvement PR #710

**Comment by waltercool** [comment]
@smurfy Please check my PR if makes sense.

@reagentoo If you can test also this as solution, would help a lot

**Comment by smurfy** [comment]
fixed with #716
